File: app/src/main/java/com/android/sellacha/fragment/OrderFragment.java
package com.android.sellacha.fragment;

import android.app.AlertDialog;
import android.app.DatePickerDialog;
import android.app.Dialog;
import android.content.DialogInterface;
import android.content.Intent;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.DatePicker;
import android.widget.TextView;

import androidx.databinding.DataBindingUtil;
import androidx.fragment.app.DialogFragment;
import androidx.recyclerview.widget.LinearLayoutManager;

import com.android.sellacha.R;
import com.android.sellacha.activity.CheckOut;
import com.android.sellacha.adapter.OrderAdapter;
import com.android.sellacha.adapter.OrderFilterSelector;
import com.android.sellacha.api.model.filterItemsDM;
import com.android.sellacha.api.response.GetAllorder.OrderItem;
import com.android.sellacha.api.response.GetAllorder.OrderResponse;
import com.android.sellacha.api.service.MainService;
import com.android.sellacha.databinding.FilterDailogBinding;
import com.android.sellacha.databinding.FragmentOrderBinding;
import com.android.sellacha.utils.AppProgressBar;
import com.google.gson.Gson;
import com.google.gson.JsonNull;

import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Calendar;
import java.util.List;
import java.util.Locale;


public class OrderFragment extends BaseFragment {
    FragmentOrderBinding binding;
    OrderFilterSelector filterNameAdapter;
    OrderAdapter orderAdapter;
    List<OrderItem> orderList = new ArrayList<>();
    filterItemsDM[] myListData;
    Dialog filterDialog;
    int filterSelector;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        if (binding == null) {
            binding = DataBindingUtil.inflate(inflater, R.layout.fragment_order, container, false);
            getAllOrder();
            //   getFilterList();
            filterDialog = new Dialog(mContext);
            binding.orderList.setLayoutManager(new LinearLayoutManager(getActivity(), LinearLayoutManager.VERTICAL, false));
            binding.searchTxt.addTextChangedListener(new TextWatcher() {
                @Override
                public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {

                }

                @Override
                public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
                    filterPlaces(charSequence.toString());
                    if (binding.searchTxt.getText().toString().isEmpty()) {
                        filterSelector(filterSelector);
                    }
                }

                @Override
                public void afterTextChanged(Editable editable) {
                }
            });
            binding.searchCloseBtn.setOnClickListener(view -> {
                binding.searchTxt.setText("");
            });
            binding.createOrder.setOnClickListener(v -> {
                startActivity(new Intent(mContext, CheckOut.class));
            });
            binding.selectFulfillment.setOnClickListener(view -> {
                filterDialog(binding.selectFulfillment);
            });
            binding.filterBtn.setOnClickListener(v -> {
                filterDialog();
            });
        }
        return binding.getRoot();
    }

    private void filterPlaces(String title) {
        ArrayList<OrderItem> filteredPlaces = new ArrayList<>();
        for (OrderItem bookModel : orderList) {
            if (bookModel.getOrderNo().toLowerCase().trim().contains(title.toLowerCase().trim())) {
                filteredPlaces.add(bookModel);
            }
        }
        orderAdapter.filteredList(filteredPlaces);
    }

    public void getAllOrder() {
        AppProgressBar.showLoaderDialog(mContext);
        MainService.getAllOrder(mContext).observe(getViewLifecycleOwner(), response -> {
            if (response == null) {
                errorSnackBar(binding.getRoot(), getString(R.string.something_wrong));
            } else {
                if (!(response.getData() instanceof JsonNull)) {
                    if (response.getData() != null) {
                        OrderResponse orderResponse = new Gson().fromJson(response.getData(), OrderResponse.class);
                        orderList.clear();
                        orderList.addAll(orderResponse.getOrders().getData());
                        getFilterList(orderResponse);
                        orderAdapter = new OrderAdapter(getContext(), orderList, 0);
                        binding.orderList.setAdapter(orderAdapter);
                    } else {
                        showAlertDialog(getString(R.string.app_name), response.getMessage(), "OK", "", DialogFragment::dismiss);
                    }
                } else {
                    errorSnackBar(binding.getRoot(), response.getMessage());
                }
            }
            AppProgressBar.hideLoaderDialog();
        });
    }

    public void getFilterOrder(int id, String status, String start, String end) {
        AppProgressBar.showLoaderDialog(mContext);
        MainService.getFilterOrder(mContext, id, status, start, end).observe(getViewLifecycleOwner(), response -> {
            if (response == null) {
                errorSnackBar(binding.getRoot(), getString(R.string.something_wrong));
            } else {
                if (!(response.getData() instanceof JsonNull)) {
                    if (response.getData() != null) {
                        OrderResponse orderResponse = new Gson().fromJson(response.getData(), OrderResponse.class);
                        orderList.clear();
                        orderList.addAll(orderResponse.getOrders().getData());
                        orderAdapter = new OrderAdapter(getContext(), orderList, 0);
                        binding.orderList.setAdapter(orderAdapter);
                    } else {
                        showAlertDialog(getString(R.string.app_name), response.getMessage(), "OK", "", DialogFragment::dismiss);
                    }
                } else {
                    errorSnackBar(binding.getRoot(), response.getMessage());
                }
            }
            AppProgressBar.hideLoaderDialog();
        });
    }

    public void filterSelector(int position) {
        filterSelector = position;
        if (position == 0) {
            getAllOrder();
        } else if (position == 1) {
            orderAdapter.notifySelection(orderList, "pending");
        } else if (position == 2) {
            orderAdapter.notifySelection(orderList, "processing");
        } else if (position == 3) {
            orderAdapter.notifySelection(orderList, "pickup");
        } else if (position == 4) {
            orderAdapter.notifySelection(orderList, "completed");
        } else if (position == 5) {
            orderAdapter.notifySelection(orderList, "canceled");
        } else if (position == 6) {
            orderAdapter.notifySelection(orderList, "archived");
        }
    }

    public void getFilterList(OrderResponse orderResponse) {
        myListData = new filterItemsDM[]{
                new filterItemsDM("All", orderResponse.getOrders().getData().size()),
                new filterItemsDM("Awaiting Processing", orderResponse.getPendings()),
                new filterItemsDM("Processing", orderResponse.getProcessing()),
                new filterItemsDM("Ready For Pickup", orderResponse.getPickup()),
                new filterItemsDM("Completed", orderResponse.getCompleted()),
                new filterItemsDM("Cancelled", orderResponse.getCanceled()),
                new filterItemsDM("Archived", orderResponse.getArchived()),
        };
        binding.filterRv.setVisibility(View.VISIBLE);
        binding.filterRv.setLayoutManager(new LinearLayoutManager(getActivity(), LinearLayoutManager.HORIZONTAL, false));
        filterNameAdapter = new OrderFilterSelector(getContext(), myListData, 0, this);
        binding.filterRv.setAdapter(filterNameAdapter);
    }

    public void filterDialog(TextView textView) {
        AlertDialog.Builder builderSingle = new AlertDialog.Builder(getContext());
        builderSingle.setTitle("Select fulfillment");
        final ArrayAdapter<String> arrayAdapter = new ArrayAdapter<String>(getActivity(), R.layout.dialog_txt_item);
        arrayAdapter.add("Awaiting processing");
        arrayAdapter.add("Processing");
        arrayAdapter.add("Ready-for-pickup");
        arrayAdapter.add("Completed");
        arrayAdapter.add("Archived");
        arrayAdapter.add("Canceled");
        builderSingle.setNegativeButton("cancel", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                dialog.dismiss();
            }
        });

        builderSingle.setAdapter(arrayAdapter, new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                String strName = arrayAdapter.getItem(which);
                textView.setText(strName);
            }
        });
        builderSingle.show();
    }

    public void paymentDialog(TextView textView) {
        AlertDialog.Builder builderSingle = new AlertDialog.Builder(getContext());
        builderSingle.setTitle("Select Payment Status");
        final ArrayAdapter<String> arrayAdapter = new ArrayAdapter<String>(getActivity(), R.layout.dialog_txt_item);
        arrayAdapter.add("Pending");
        arrayAdapter.add("Complete");
        arrayAdapter.add("Cancel");
        builderSingle.setNegativeButton("cancel", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                dialog.dismiss();
            }
        });

        builderSingle.setAdapter(arrayAdapter, new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                String strName = arrayAdapter.getItem(which);
                textView.setText(strName);
            }
        });
        builderSingle.show();
    }

    public void filterDialog() {
        final int[] payment = {0};
        FilterDailogBinding sDialog = DataBindingUtil.inflate(LayoutInflater.from(mContext), R.layout.filter_dailog, null, false);
        filterDialog.setContentView(sDialog.getRoot());
        filterDialog.setCancelable(true);
        filterDialog.setCanceledOnTouchOutside(true);
        sDialog.startingDateTxt.setOnClickListener(v -> {
            datePicker(sDialog.startingDateTxt);
        });
        sDialog.endingDateTxt.setOnClickListener(v -> {
            datePicker(sDialog.endingDateTxt);
        });
        sDialog.fulfilmentstatusTxt.setOnClickListener(v -> {
            filterDialog(sDialog.fulfilmentstatusTxt);
        });
        sDialog.fulfilmentstatusTxt.setOnClickListener(v -> {
            filterDialog(sDialog.fulfilmentstatusTxt);
        });
        sDialog.paymentStatusTxt.setOnClickListener(v -> {
            paymentDialog(sDialog.paymentStatusTxt);
        });
        sDialog.cancelBtn.setOnClickListener(v -> {
            sDialog.startingDateTxt.setText("");
            sDialog.endingDateTxt.setText("");
            sDialog.fulfilmentstatusTxt.setText("");
            sDialog.paymentStatusTxt.setText("");
        });
        sDialog.okBtn.setOnClickListener(v -> {
            filterDialog.dismiss();
            if (sDialog.paymentStatusTxt.getText().toString().equals("Pending")) {
                payment[0] = 2;
            } else if (sDialog.paymentStatusTxt.getText().toString().equals("Complete")) {
                payment[0] = 1;
            } else if (sDialog.paymentStatusTxt.getText().toString().equals("Cancel")) {

            }
            getFilterOrder(payment[0], sDialog.fulfilmentstatusTxt.toString(), sDialog.startingDateTxt.toString(), sDialog.endingDateTxt.toString());
        });
        filterDialog.getWindow().setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT);
        filterDialog.getWindow().setGravity(Gravity.CENTER);
        filterDialog.getWindow().setBackgroundDrawable(new ColorDrawable(android.graphics.Color.TRANSPARENT));
        filterDialog.show();
    }

    public void datePicker(TextView textView) {
        final Calendar myCalendar = Calendar.getInstance();
        DatePickerDialog.OnDateSetListener date = new DatePickerDialog.OnDateSetListener() {
            @Override
            public void onDateSet(DatePicker view, int year, int monthOfYear, int dayOfMonth) {
                // TODO Auto-generated method stub
                myCalendar.set(Calendar.YEAR, year);
                myCalendar.set(Calendar.MONTH, monthOfYear);
                myCalendar.set(Calendar.DAY_OF_MONTH, dayOfMonth);
                String myFormat = "yyyy-MM-dd"; // your format
                SimpleDateFormat sdf = new SimpleDateFormat(myFormat, Locale.getDefault());
                textView.setText(sdf.format(myCalendar.getTime()));
            }
        };
        new DatePickerDialog(mContext, R.style.datepicker, date, myCalendar.get(Calendar.YEAR), myCalendar.get(Calendar.MONTH), myCalendar.get(Calendar.DAY_OF_MONTH)).show();

    }

}

File: app/src/main/java/com/android/sellacha/fragment/OrderFragment.kt
package com.android.sellacha.fragment

import android.app.AlertDialog
import android.app.DatePickerDialog
import android.app.DatePickerDialog.OnDateSetListener
import android.app.Dialog
import android.content.Intent
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.util.Log
import android.view.Gravity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView
import androidx.databinding.DataBindingUtil
import androidx.navigation.Navigation
import androidx.recyclerview.widget.LinearLayoutManager
import com.android.sellacha.R
import com.android.sellacha.activity.CheckOut
import com.android.sellacha.adapter.OrderAdapter
import com.android.sellacha.adapter.OrderFilterSelector
import com.android.sellacha.api.ApiResponse
import com.android.sellacha.api.model.filterItemsDM
import com.android.sellacha.api.response.GetAllorder.OrderItem
import com.android.sellacha.api.response.GetAllorder.OrderResponse
import com.android.sellacha.api.service.MainService
import com.android.sellacha.databinding.FilterDailogBinding
import com.android.sellacha.databinding.FragmentOrderBinding
import com.android.sellacha.dialog.AppDialog
import com.android.sellacha.utils.AppProgressBar
import com.google.gson.Gson
import com.google.gson.JsonNull
import java.text.SimpleDateFormat
import java.util.*

class OrderFragment : BaseFragment() {
    var binding: FragmentOrderBinding? = null
    var filterNameAdapter: OrderFilterSelector? = null
    var orderAdapter: OrderAdapter? = null
    var orderList: MutableList<OrderItem> = ArrayList()
    private lateinit var myListData: Array<filterItemsDM>
    var filterDialog: Dialog? = null
    var filterSelector = 0
    var selectionPosition = 0
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        if (binding == null) {
            binding = DataBindingUtil.inflate(inflater, R.layout.fragment_order, container, false)
            allOrder
            //   getFilterList();
            filterDialog = Dialog(mContext)
            binding!!.orderList.layoutManager = LinearLayoutManager(
                activity, LinearLayoutManager.VERTICAL, false)
            binding!!.searchTxt.addTextChangedListener(object : TextWatcher {
                override fun beforeTextChanged(
                    charSequence: CharSequence,
                    i: Int,
                    i1: Int,
                    i2: Int
                ) {
                }

                override fun onTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {
                    filterPlaces(charSequence.toString())
                    if (binding!!.searchTxt.text.toString().isEmpty()) {
                        filterSelector(filterSelector)
                    }
                }

                override fun afterTextChanged(editable: Editable) {}
            })

            binding!!.searchCloseBtn.setOnClickListener {
                binding!!.searchTxt.setText("")
            }
            binding!!.createOrder.setOnClickListener {
                Navigation.findNavController(binding!!.root).navigate(R.id.CreateOrderShow)
              //  startActivity(Intent(mContext, CheckOut::class.java))
            }
            binding!!.selectFulfillment.setOnClickListener {
                filterDialog(binding!!.selectFulfillment)

            }

            binding!!.btnSubmit.setOnClickListener {
                when (selectionPosition) {
//                    0 -> {
//                        allOrder
//                    }
                    0 -> {
                        orderAdapter!!.notifySelection(orderList, "pending")
                    }
                    1 -> {
                        orderAdapter!!.notifySelection(orderList, "processing")
                    }
                    2 -> {
                        orderAdapter!!.notifySelection(orderList, "pickup")
                    }
                    3 -> {
                        orderAdapter!!.notifySelection(orderList, "completed")
                    }
                    4 -> {
                        orderAdapter!!.notifySelection(orderList, "archived")
                    }
                    5 -> {
                        orderAdapter!!.notifySelection(orderList, "canceled")

                    }
                }
            }

            binding!!.filterBtn.setOnClickListener { filterDialog() }
        }
        return binding!!.root
    }

    private fun filterPlaces(title: String) {
        val filteredPlaces = ArrayList<OrderItem>()
        for (bookModel in orderList) {
            if (bookModel.orderNo.lowercase(Locale.getDefault()).trim { it <= ' ' }
                    .contains(title.lowercase(Locale.getDefault()).trim { it <= ' ' })) {
                filteredPlaces.add(bookModel)
            }
        }
        orderAdapter!!.filteredList(filteredPlaces)
    }

    private val allOrder: Unit
        get() {
            AppProgressBar.showLoaderDialog(mContext)
            MainService.getAllOrder(mContext)
                .observe(viewLifecycleOwner) { response: ApiResponse? ->
                    if (response == null) {
                        errorSnackBar(binding!!.root, getString(R.string.something_wrong))
                    } else {
                        if (response.data !is JsonNull) {
                            if (response.data != null) {
                                val orderResponse =
                                    Gson().fromJson(response.data, OrderResponse::class.java)
                                orderList.clear()
                                orderList.addAll(orderResponse.orders.data)
                                getFilterList(orderResponse)
                                orderAdapter = context?.let { OrderAdapter(it, orderList, 0)
                                }
                                binding!!.orderList.adapter = orderAdapter
                            } else {
                                showAlertDialog(getString(R.string.app_name), response.message, "OK", "") { obj: AppDialog -> obj.dismiss() }
                            }
                        } else {
                            errorSnackBar(binding!!.root, response.message)
                        }
                    }
                    AppProgressBar.hideLoaderDialog()
                }
        }

    private fun getFilterOrder(id: Int, status: String?, start: String?, end: String?) {
        AppProgressBar.showLoaderDialog(mContext)
        MainService.getFilterOrder(mContext, id, status, start, end)
            .observe(viewLifecycleOwner) { response: ApiResponse? ->
                if (response == null) {
                    errorSnackBar(binding!!.root, getString(R.string.something_wrong))
                } else {
                    if (response.data !is JsonNull) {
                        if (response.data != null) {
                            val orderResponse =
                                Gson().fromJson(response.data, OrderResponse::class.java)
                            orderList.clear()
                            orderList.addAll(orderResponse.orders.data)
                            orderAdapter = context?.let { OrderAdapter(it, orderList, 0) }
                            binding!!.orderList.adapter = orderAdapter
                        } else {
                            showAlertDialog(
                                getString(R.string.app_name),
                                response.message,
                                "OK",
                                ""
                            ) { obj: AppDialog -> obj.dismiss() }
                        }
                    } else {
                        errorSnackBar(binding!!.root, response.message)
                    }
                }
                AppProgressBar.hideLoaderDialog()
            }
    }

    fun filterSelector(position: Int) {
        filterSelector = position
        when (position) {
            0 -> {
                allOrder
            }
            1 -> {
                orderAdapter!!.notifySelection(orderList, "pending")
            }
            2 -> {
                orderAdapter!!.notifySelection(orderList, "processing")
            }
            3 -> {
                orderAdapter!!.notifySelection(orderList, "pickup")
            }
            4 -> {
                orderAdapter!!.notifySelection(orderList, "completed")
            }
            5 -> {
                orderAdapter!!.notifySelection(orderList, "canceled")
            }
            6 -> {
                orderAdapter!!.notifySelection(orderList, "archived")
            }
        }
    }

    private fun getFilterList(orderResponse: OrderResponse) {
        myListData = arrayOf(
            filterItemsDM("All", orderResponse.orders.data.size),
            filterItemsDM("Awaiting Processing", orderResponse.pendings),
            filterItemsDM("Processing", orderResponse.processing),
            filterItemsDM("Ready For Pickup", orderResponse.pickup),
            filterItemsDM("Completed", orderResponse.completed),
            filterItemsDM("Cancelled", orderResponse.canceled),
            filterItemsDM("Archived", orderResponse.archived)
        )


        binding!!.filterRv.visibility = View.VISIBLE
        binding!!.filterRv.layoutManager = LinearLayoutManager(activity, LinearLayoutManager.HORIZONTAL, false)
        filterNameAdapter = OrderFilterSelector(context, myListData, 0, this)
        binding!!.filterRv.adapter = filterNameAdapter
    }

    private fun filterDialog(textView: TextView) {
        val builderSingle = AlertDialog.Builder(context)
        builderSingle.setTitle("Select fulfillment")

        val arrayAdapter = ArrayAdapter<String>(activity!!, R.layout.dialog_txt_item)

        arrayAdapter.add("Awaiting processing")
        arrayAdapter.add("Processing")
        arrayAdapter.add("Ready-for-pickup")
        arrayAdapter.add("Completed")
        arrayAdapter.add("Archived")
        arrayAdapter.add("Canceled")
        builderSingle.setNegativeButton("cancel") { dialog, _ -> dialog.dismiss()

        }
        builderSingle.setAdapter(arrayAdapter) { pos, which ->
            val strName = arrayAdapter.getItem(which)
            selectionPosition =which
            Log.e("posiction",which.toString())
            textView.text = strName
        }
        builderSingle.show()
    }

    private fun paymentDialog(textView: TextView) {
        val builderSingle = AlertDialog.Builder(context)
        builderSingle.setTitle("Select Payment Status")
        val arrayAdapter = ArrayAdapter<String>(activity!!, R.layout.dialog_txt_item)
        arrayAdapter.add("Pending")
        arrayAdapter.add("Complete")
        arrayAdapter.add("Cancel")
        builderSingle.setNegativeButton("cancel") { dialog, _ -> dialog.dismiss() }
        builderSingle.setAdapter(arrayAdapter) { _, which ->
            val strName = arrayAdapter.getItem(which)
            textView.text = strName
        }
        builderSingle.show()
    }

    private fun filterDialog() {
        val payment = intArrayOf(0)
        val sDialog = DataBindingUtil.inflate<FilterDailogBinding>(
            LayoutInflater.from(mContext),
            R.layout.filter_dailog,
            null,
            false
        )
        filterDialog!!.setContentView(sDialog.root)
        filterDialog!!.setCancelable(true)
        filterDialog!!.setCanceledOnTouchOutside(true)
        sDialog.startingDateTxt.setOnClickListener { datePicker(sDialog.startingDateTxt) }
        sDialog.endingDateTxt.setOnClickListener { datePicker(sDialog.endingDateTxt) }
        sDialog.fulfilmentstatusTxt.setOnClickListener {
            filterDialog(sDialog.fulfilmentstatusTxt)
        }
        sDialog.fulfilmentstatusTxt.setOnClickListener { filterDialog(sDialog.fulfilmentstatusTxt) }
        sDialog.paymentStatusTxt.setOnClickListener { paymentDialog(sDialog.paymentStatusTxt) }
        sDialog.cancelBtn.setOnClickListener {
            sDialog.startingDateTxt.text = ""
            sDialog.endingDateTxt.text = ""
            sDialog.fulfilmentstatusTxt.text = ""
            sDialog.paymentStatusTxt.text = ""
        }
        sDialog.okBtn.setOnClickListener {
            filterDialog!!.dismiss()
            if (sDialog.paymentStatusTxt.text.toString() == "Pending") {
                payment[0] = 2
            } else if (sDialog.paymentStatusTxt.text.toString() == "Complete") {
                payment[0] = 1
            } else if (sDialog.paymentStatusTxt.text.toString() == "Cancel") {
            }
            getFilterOrder(
                payment[0],
                sDialog.fulfilmentstatusTxt.toString(),
                sDialog.startingDateTxt.toString(),
                sDialog.endingDateTxt.toString()
            )
        }
        filterDialog!!.window!!.setLayout(
            ViewGroup.LayoutParams.MATCH_PARENT,
            ViewGroup.LayoutParams.MATCH_PARENT
        )
        filterDialog!!.window!!.setGravity(Gravity.CENTER)
        filterDialog!!.window!!.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
        filterDialog!!.show()
    }

    private fun datePicker(textView: TextView) {
        val myCalendar = Calendar.getInstance()
        val date =
            OnDateSetListener { _, year, monthOfYear, dayOfMonth -> // TODO Auto-generated method stub
                myCalendar[Calendar.YEAR] = year
                myCalendar[Calendar.MONTH] = monthOfYear
                myCalendar[Calendar.DAY_OF_MONTH] = dayOfMonth
                val myFormat = "yyyy-MM-dd" // your format
                val sdf = SimpleDateFormat(myFormat, Locale.getDefault())
                textView.text = sdf.format(myCalendar.time)
            }
        DatePickerDialog(
            mContext,
            R.style.datepicker,
            date,
            myCalendar[Calendar.YEAR],
            myCalendar[Calendar.MONTH],
            myCalendar[Calendar.DAY_OF_MONTH]
        ).show()
    }
}